core/src/runtime/hooks.rs

```diff
@@ -0,0 +1,119 @@
+use core::convert::TryInto;
+use std::collections::HashMap;
+
+use k256::ecdsa::{RecoveryId, Signature, VerifyingKey};
+use k256::elliptic_curve::ops::Invert;
+
+use super::Runtime;
+
+type BoxedHook<'a> = Box<dyn Fn(HookEnv, &[u8]) -> Vec<Vec<u8>> + 'a>;
+
+/// The file descriptor through which to access `hook_ecrecover`.
+pub const FD_ECRECOVER_HOOK: u32 = 5;
+// Note: To ensure any `fd` value is synced with `zkvm/precompiles/src/io.rs`,
```

Comment:
why is this a // comment, use "///" and put it above

Reply:
This comment does not have to do with `FD_ECRECOVER_HOOK` specifically, and would confuse people who are merely using that constant. I put it where other people may eventually add more constants so that they read it first.

Reply:
Moved it